Mercury-Store/Mercury-Store/Home/Views/BrandDetails/Cells/BrandProductsCollectionViewCell.swift

```diff
@@ -38,6 +36,13 @@ class BrandProductsCollectionViewCell: UICollectionViewCell {
             productForBrandImage.downloadImage(url: url , placeholder: UIImage(named: "placeholder"), imageIndicator: .gray, completion: nil)
             productForBrandName.text = item.title
             productForBrandPrice.text = item.variants[0].price
+            favouriteButton.favouriteState(state: viewModel.getFavouriteState(productID: item.id))
+            favouriteButton.rx.tap.subscribe(onNext: { [ weak self ] in
+                let savedValue = SavedProductItem(productID: Decimal(item.id), productTitle: item.title, productImage: item.image.src, productPrice: Double(item.variants[0].price )! , productQTY: 0, producrState: productStates.favourite.rawValue)
+                let state = self?.viewModel.toggleFavourite(product: savedValue)
+                self?.favouriteButton.favouriteState(state: state!)
```

Comment:
No need to force unwrap.. Use `guard let self = self else { return }` at the beginning of the scope.